src/Views/Main/Main.js

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useState } from 'react';
+import './Main.css';
+import { fetchPokemon, fetchTypes, filterTypes } from '../../services/pokemon/fetchpokemon';
+import Filter from '../../components/Controls/Filter/Filter';
+import SearchBar from '../../components/Controls/Search/Search';
+import Order from '../../components/Controls/Order/Order';
+
+export default function Main() {
+
+  const [pokemon, setPokemon] = useState([]);
+  const [types, setTypes] = useState([]);
+  const [selectedType, setSelectedType] = useState('all');
+  const [error, setError] = useState('');
+  const [loading, setLoading] = useState(true);
+  const [search, setSearch] = useState('');
+  const [order, setOrder] = useState('');
+
+  useEffect(() =>{
+    const fetchData = async () => {
+      const typesData = await fetchTypes();
+      setTypes(['all', ...typesData]);
+      
+    };
+    fetchData();
+  }, []);
+
+  useEffect(() => {
+    const fetchData = async () => {
+
+      try {
+        const data = await fetchPokemon();
+        setPokemon(data);
+        setLoading(false);
+      } catch (error) {
+        setError(error.message);
+      }
+    };
+    fetchData();
+  }, []);
+
+  useEffect(() =>{
+    const chosenType = async () => {
+      const matchedTypes = await filterTypes(selectedType, null, order);
+      setPokemon(matchedTypes);
```

Comment:
Here we can also pass in search to avoid the reset when selectType and Order change.